test/uk/gov/hmrc/tai/connectors/PersonConnectorSpec.scala

```diff
@@ -32,31 +31,31 @@ class PersonConnectorSpec extends BaseSpec {
 
   "person method" must {
 
-    "return a Person model instance, wrapped in a TaiSuccessResponse" when {
+    "return a Person model instance" when {
       "the http call returns successfully" in {
         when(httpHandler.getFromApiV2(Matchers.eq(s"/fakeUrl/tai/${nino.nino}/person"))(any()))
           .thenReturn(Future.successful(apiResponse(person)))
-        val result = Await.result(sut.person(nino), 5 seconds)
-        result mustBe TaiSuccessResponseWithPayload(person)
+
+        val result = Await.result(sut.person(nino), 5.seconds)
+        result mustBe person
       }
     }
 
-    "return a TaiNotFoundResponse" when {
+    "return a Failed Future" when {
       "the http call returns a not found exception" in {
         when(httpHandler.getFromApiV2(Matchers.eq(s"/fakeUrl/tai/${nino.nino}/person"))(any()))
           .thenReturn(Future.failed(new NotFoundException("downstream not found")))
-        val result = Await.result(sut.person(nino), 5 seconds)
-        result mustBe TaiNotFoundResponse("downstream not found")
+        assertThrows[NotFoundException] {
+          Await.result(sut.person(nino), 5.seconds)
```

Comment:
Why would line 49 throw an exception?

Reply:
Rather than wrapping things in TaiReponses that need to be pattern match. We're letting the Future fail. When awaiting for a future, if this one fails, the exception would be thrown.

So rather than having Future[TaiReponse] now we'd have: Future[Person] and the error channel is left to propagate any internal exceptions. Vivek is working in the error handling so it might be in the future rather than using the Future error channel (Future.failed) we use something like Future[Either[Error, Person]]

Reply:
Okay I think I'm with it, this stuff is confusing man :'(